hamilton/function_modifiers/adapters.py

```diff
@@ -259,6 +260,7 @@ def filter_function(_inject_parameter=inject_parameter, **kwargs):
             # to have this weird DAG shape. For now, this solves the problem, and this is an internal component of the API
             # so we're good to go
             namespace=((namespace, "select_data") if namespace else ()),  # We want no namespace
+            originating_functions=(filter_function,),
```

Comment:
Same as above, this is a local lambda, and won't tell us anything interesting. This should already be added.